### PR TITLE
Add initial support for SCP

### PR DESCRIPTION
**Pull request by dainnilsson**
Implements support for Security Domain session. 

- Key management allows clients to import, replace and delete SCP03 and SCP11 keys and certificates.
- Application sessions (PIV, OpenPGP, OATH, Ctap2, YubiOTP) have new constructors with SCP parameters to initialize secure communication over smart-card connection.
- Updated device tests verify functionality of the management features as well test YubiKey applications over a secure SCP11b connection.

### CODE REVIEW

File: testing/src/main/java/com/yubico/yubikit/testing/piv/PivTestState.java [hunk 1]
```diff
@@ -16,15 +16,151 @@
...
+    private static final byte[] COMPLEX_MANAGEMENT_KEY = new byte[]{
+            0x01, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x01, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x01, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    };
+
+    public final boolean isFipsApproved;
+    public char[] defaultPin;
+    public char[] defaultPuk;
+    public byte[] defaultManagementKey;
```
Comment (dainnilsson) [inline testing/src/main/java/com/yubico/yubikit/testing/piv/PivTestState.java:61]: These should probably be renamed as they aren't always the default values. Maybe "initialPin" or similar?

File: core/src/main/java/com/yubico/yubikit/core/smartcard/scp/Scp11KeyParams.java [hunk 2]
```diff
@@ -70,4 +70,13 @@ public Scp11KeyParams(KeyRef keyRef, PublicKey pkSdEcka) {
...
     }
+
+    public List<X509Certificate> getCertificates() {
+        return certificates;
+    }
+
+    @Nullable
+    public KeyRef getOceKeyRef() {
+        return oceKeyRef;
+    }
```
Comment (dainnilsson) [inline core/src/main/java/com/yubico/yubikit/core/smartcard/scp/Scp11KeyParams.java:81]: Are these only added for testing? Is there anyway we can do this without having to make additions to the public API?

Or do we think they do actually make sense to have even outside of testing?

File: piv/src/main/java/com/yubico/yubikit/piv/PivSession.java [hunk 3]
```diff
@@ -1458,7 +1457,7 @@ private int getRetriesFromCode(int statusCode) {
         if (statusCode == SW.AUTH_METHOD_BLOCKED) {
             return 0;
         }
-        if (version.isAtLeast(1, 0, 0) && version.isLessThan(1, 0, 4)) {
+        if (version.isLessThan(1, 0, 4)) {
```
Comment (dainnilsson) [inline piv/src/main/java/com/yubico/yubikit/piv/PivSession.java:1460]: Why is this logic changed, was the old behavior incorrect?

Comment (AdamVe) [reply piv/src/main/java/com/yubico/yubikit/piv/PivSession.java:1460]: This is not part of the PR, but was used during development.

File: core/src/main/java/com/yubico/yubikit/core/smartcard/scp/Scp11KeyParams.java (hunk 2 above)
Comment (AdamVe) [reply core/src/main/java/com/yubico/yubikit/core/smartcard/scp/Scp11KeyParams.java]: I think we can go without these for now. I removed them and updated the tests.

File: testing/src/main/java/com/yubico/yubikit/testing/piv/PivTestState.java (hunk 1 above)
Comment (AdamVe) [reply testing/src/main/java/com/yubico/yubikit/testing/piv/PivTestState.java]: fixed!

**Pull request merged**